### PR TITLE
Implement Slack deploy button; Add documentation/cleanups

### PR DESCRIPTION
**Pull request by MichaelDeng03**
- IMPORTANT: Adds starting prompt to Developer context in communicative_dehallucination. 
- Add slack event handler
  - Responds to messages posted in #github-logs with a button
- Add slack button handler
  - All slack interactive modalities, upon interaction, will deploy latest image in ECR to prod. This includes the button. 
- Add pretty print of Response objects
- Fix prompt indentation
- Add flexibility to AwsTool._deploy_image (hardware + url)

QOL: 
- Rename webapp-main -> webapp-homepage
- Add documentation to slack button

srry for huge pr :(

### CODE REVIEW

File: concrete/operator_responses.py [hunk 1]
```diff
@@ -13,15 +13,34 @@ class MyClass(BaseModel):
...
+import json
 from typing import List
 
 from pydantic import BaseModel, Field
 
 
 class Response(BaseModel):
     def __str__(self):
-        return dumps(self.model_dump(), indent=2)
+        def parse_json_strings(obj):
```
Comment (kgang) [inline concrete/operator_responses.py:24]: Add a couple comments on what's going on and why

File: concrete/orchestrator.py [hunk 2]
```diff
@@ -157,10 +159,10 @@ async def communicative_dehallucination(
             - summary (str): A concise summary of what has been achieved.
     """
 
-    context = dedent(
-        f"""Previous Components summarized:\n{summary}
-    Current Component: {component}"""
-    )
+    context = f"""Previously Implemented Components summarized:\n{summary}
```
Comment (kgang) [inline concrete/orchestrator.py:162]: What was wrong with dedent

File: concrete/tools.py [hunk 3]
```diff
@@ -266,11 +266,20 @@ def _poll_image_status(cls, repo_name: str) -> bool:
 
     @classmethod
     def _deploy_image(
-        cls, image_uri: str, custom_name: Optional[str] = None, cpu: int = 256, memory: int = 512
+        cls,
+        image_uri: str,
+        custom_name: Optional[str] = None,
+        cpu: Optional[int] = None,
+        memory: Optional[int] = None,
+        listener_rule: Optional[dict] = None,
```
Comment (kgang) [inline concrete/tools.py:274]: Niceu

**Review by kgang — COMMENTED** [review]
Looking good!

File: concrete/tools.py [hunk 4]
```diff
@@ -354,8 +371,8 @@ def _deploy_image(
                     },
                 }
             ],
-            cpu=str(cpu),
-            memory=str(memory),
+            cpu=str(cpu) if cpu else "256",
+            memory=str(memory) if memory else "512",
```
[the remark this answers is not in the document]
Comment (MichaelDeng03) [reply concrete/tools.py:375]: I had that earlier, but str(None) evaluates to True

File: webapp/demo/server.py [hunk 5]
```diff
@@ -79,24 +96,77 @@ def deploy_images(response_url: str):
...
+    background_tasks.add_task(_deploy_to_prod, response_url=payload['response_url'])
 
     return payload
 
 
+def _post_button():
+    """
+    Posts a button to #github-logs under slack bot user.
+    """
+    # TODO Add background handling of this function to return response to slack immediately
```
[the remark this answers is not in the document]
Comment (MichaelDeng03) [reply webapp/demo/server.py:122]: Oh yeah, I forgot about that - in fact, I'm already using that solution.

File: concrete/tools.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (kgang) [reply concrete/tools.py:375]: Got it [](edited-with-Axolo)

File: concrete/orchestrator.py (hunk 2 above)
Comment (MichaelDeng03) [reply concrete/orchestrator.py:162]: When I was examining CLIClient outputs there were still a ton of malformed prompts, not sure if we were just missing dedents somewhere, or if dedent didn't work as we expected. I can look back into that, but I'm confident the current solution has correct indentation.

Comment (kgang) [reply concrete/orchestrator.py:162]: Sounds good [](edited-with-Axolo)

**Review by kgang — APPROVED** [review]
LGtM

**Pull request merged**